File: backend/payment/src/main/java/com/commonsengine/payment/infrastructure/persistence/LedgerEventRepository.java
package com.commonsengine.payment.infrastructure.persistence;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

/**
 * 账本事件 Repository。
 */
@Repository
public interface LedgerEventRepository extends JpaRepository<LedgerEventEntity, Long> {

    List<LedgerEventEntity> findByTransactionId(String transactionId);

    long countByEventType(String eventType);
}

File: backend/payment/src/main/kotlin/com/commonsengine/payment/infrastructure/persistence/LedgerEventRepository.kt
package com.commonsengine.payment.infrastructure.persistence

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface LedgerEventRepository : JpaRepository<LedgerEventEntity, Long> {

    fun findByTransactionId(transactionId: String): List<LedgerEventEntity>

    fun countByEventType(eventType: String): Long
}
